lib/routemap.c

```diff
@@ -2747,7 +2747,9 @@ static int route_map_dep_update(struct hash *dephash, const char *dep_name,
 		memset(&tmp_dep_data, 0, sizeof(struct route_map_dep_data));
 		tmp_dep_data.rname = rname;
 		dep_data = hash_lookup(dep->dep_rmap_hash, &tmp_dep_data);
-		dep_data->refcnt--;
+
+		if (dep_data->refcnt)
```

Comment:
I think you meant `if (dep_data)`? (otherwise you are already accessing the NULL pointer)

How about the other `dep_data` accesses right after this?

Reply:
yeah, `dep_data`, mixed :)

Reply:
@rzalamena I added a test for a whole dep_data